frontend/src/metabase/public/containers/PublicAction/PublicActionLoader.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useCallback, useState } from "react";
+import { connect } from "react-redux";
+
+import { useOnMount } from "metabase/hooks/use-on-mount";
+import { useSafeAsyncFunction } from "metabase/hooks/use-safe-async-function";
+import { setErrorPage } from "metabase/redux/app";
+import { PublicApi } from "metabase/services";
+
+import type { WritebackAction } from "metabase-types/api";
+import type { AppErrorDescriptor } from "metabase-types/store";
+
+import EmbedFrame from "../../components/EmbedFrame";
+import PublicAction from "./PublicAction";
+import {
+  LoadingAndErrorWrapper,
+  ContentContainer,
+} from "./PublicAction.styled";
+
+interface OwnProps {
+  params: { uuid: string };
+}
+
+interface DispatchProps {
+  setErrorPage: (error: AppErrorDescriptor) => void;
+}
+
+type Props = OwnProps & DispatchProps;
+
+const mapDispatchToProps = {
+  setErrorPage,
+};
+
+function PublicActionLoader({ params, setErrorPage }: Props) {
+  const [action, setAction] = useState<WritebackAction | null>(null);
+  const fetchAction = useSafeAsyncFunction(PublicApi.action);
+
+  useOnMount(() => {
```

Comment:
I wonder if we are encouraging people to use this hook instead of directly use `useEffect` with an empty array as a dependency list?

Reply:
I'm not sure we have a strict agreement here, but I find it more explicit